databases/databases/databases_and_sql.md

```diff
@@ -139,19 +139,24 @@ If you never quite get to the point where you're comfortable with the really adv
 
 The next step, once you've had a chance to practice this all in the project, is to apply it to Rails with Active Record.  You'll quickly find that Active Record makes your life much, much, much better.  Just don't forget about ol' SQL when you've moved onto those better and brighter things, okay?
 
-### Knowledge Checks
+### Knowledge check
 This section contains questions for you to check your understanding of this lesson on your own. If you’re having trouble answering a question, click it and review the material it links to.
 
--   [What is the difference between a foreign key and a primary key?](#foreign-key)
--   [Where is the setup information for your database stored?](#schema)
--   [What are the important parts of a SQL command?](#command-parts)
--   [Which SQL statement is associated with "Read" from the CRUD acronym?](#sql-read)
--   [Which `JOIN` statement keeps only the rows from both tables where they match up?](#inner-join)
--   [How do you use an aggregate function?](#aggregate-function)
--   [In which situation would you use the `HAVING` function?](#having-function)
--   [Why can't I just use Ruby to process my database data?](#sql-is-faster-than-ruby)
-
-### Additional Resources
+- [Introduction](#introduction)
+  - [A note on resources](#a-note-on-resources)
+- [Lesson overview](#lesson-overview)
+- [The world's fastest semi-complete explanation of SQL](#the-worlds-fastest-semi-complete-explanation-of-sql)
+  - [Setting stuff up](#setting-stuff-up)
+  - [Mucking around with data](#mucking-around-with-data)
+  - [Mashing tables together](#mashing-tables-together)
+  - [Using functions to aggregate your data](#using-functions-to-aggregate-your-data)
+- [SQL is faster than Ruby!](#sql-is-faster-than-ruby)
+- [Assignment](#assignment)
+- [Conclusion](#conclusion)
+- [Knowledge check](#knowledge-check)
+- [Additional resources](#additional-resources)
+
+### Additional resources
```

Comment:
```suggestion
- [What is the difference between a foreign key and a primary key?](#foreign-key)
- [Where is the setup information for your database stored?](#schema)
- [What are the important parts of a SQL command?](#command-parts)
- [Which SQL statement is associated with "Read" from the CRUD acronym?](#sql-read)
- [Which `JOIN` statement keeps only the rows from both tables where they match up?](#inner-join)
- [How do you use an aggregate function?](#aggregate-function)
- [In which situation would you use the `HAVING` function?](#having-function)
- [Why can't I just use Ruby to process my database data?](#sql-is-faster-than-ruby)

 ### Additional Resources
```

You can click `commit suggestion` if you want to directly commit my proposed reversal, which will avoid you having to locally edit it again and pushing.